Plutarch/ByteString.hs

```diff
@@ -37,6 +47,33 @@ phexByteStr = punsafeConstant . PLC.Some . PLC.ValueOf PLC.DefaultUniByteString
 pbyteStr :: ByteString -> Term s PByteString
 pbyteStr = punsafeConstant . PLC.Some . PLC.ValueOf PLC.DefaultUniByteString
 
+-----------------------------------------------------------
+-- The following functions should be import qualified. --
+-----------------------------------------------------------
+
+-- | Prepend a byte, represented by a non negative 'PInteger', to a 'PBytestring'.
+pcons :: Term s (PInteger :--> PByteString :--> PByteString)
+pcons = punsafeBuiltin PLC.ConsByteString
+
+{- | Slice a 'PByteString' with given start and end indices.
+
+>>> (pslice # 1 # 3 phexByteStr "4102afde5b2a") #== phexByteStr "02afde"
+-}
+pslice :: Term s (PInteger :--> PInteger :--> PByteString :--> PByteString)
+pslice = punsafeBuiltin PLC.SliceByteString
+
+-- | Find the length of a 'PByteString'.
+plength :: Term s (PByteString :--> PInteger)
+plength = punsafeBuiltin PLC.LengthOfByteString
+
+-- | 'PByteString' indexing function.
+pindex :: Term s (PByteString :--> PInteger :--> PInteger)
+pindex = punsafeBuiltin PLC.IndexByteString
+
+-- | Pack a list of bytes into a 'PByteString' term.
+ppack :: [Word8] -> Term s PByteString
+ppack = punsafeConstant . PLC.Some . PLC.ValueOf PLC.DefaultUniByteString . BS.pack
```

Comment:
`ppack = pbyteStr . BS.pack`, I don't think we need this.

Reply:
Hmm, true - actually. Going to remove it.